src/core/mac/mac.cpp

```diff
@@ -982,9 +989,10 @@ void Mac::ProcessTransmitSecurity(TxFrame &aFrame, bool aProcessAesCcm)
         {
             aFrame.SetFrameCounter(keyManager.GetMacFrameCounter());
             keyManager.IncrementMacFrameCounter();
-            aFrame.SetKeyId((keyManager.GetCurrentKeySequence() & 0x7f) + 1);
         }
 
+        // For MAC key ID mode 1, the AES CCM* is done at SubMac or Radio if supported
+        processTransmitAesCcm = false;
```

Comment:
Verify the radio supports transmit encryption?

Reply:
No need. It will be passed down to `SubMac` and check radio capability there.

Reply:
ExitNow();